### PR TITLE
Update Xcode to 15.2

### PR DESCRIPTION
**Pull request by karim-alweheshy**
Updated CI runners to [macos-14](https://github.com/actions/runner-images/blob/main/images/macos/macos-14-Readme.md)
Update Xcode to 15.2.0 from 14.2.0
Update min iOS version to 12.0 from 10.0 and 11.0 and remove Host-App pre 12.0
Updated iOS sdk to 17.2 and devices to sims from the runner
Update swift_version from 5.1 to 5.9
Update xcodeproj installer, as it was pointing to a static lib for dynamic lib compilation and fixed path to remove //
Updated compile frameworks to the correct arch ios-sim-arm64-x86_64
Updated third party libs to be on the right arch ios-sim-arm64-x86_64

old PR: https://github.com/bazel-ios/rules_ios/pull/818

### CODE REVIEW

File: tools/xcodeproj_shims/install.sh
Comment (luispadron) [inline tools/xcodeproj_shims/install.sh:13]: Do you know why this changed?

```suggestion
    input_options=("bazel-bin/$BAZEL_BIN_SUBDIR/${TARGET_NAME}.framework")
```

File: tools/xcodeproj_shims/install.sh
Comment (luispadron) [inline tools/xcodeproj_shims/install.sh:15]:
```suggestion
        input_options=("bazel-bin/$BAZEL_BIN_SUBDIR/${TARGET_NAME}.framework.zip")
```

**Comment by luispadron** [comment]
@karim-alweheshy I think you'll need to re-update the fixtures with the reverted change to the path

**Comment by karim-alweheshy** [comment]
> @karim-alweheshy I think you'll need to re-update the fixtures with the reverted change to the path

Done @luispadron

File: tools/xcodeproj_shims/install.sh
Comment (luispadron) [inline tools/xcodeproj_shims/install.sh:1]: I don't really understand this change so I think @thiagohmcruz @jerrymarino weigh in on this one since they depend on the legacy project stuff.

**Review by luispadron — APPROVED** [review]
Left a comment for others to look at, the rest of the changes look good though thanks for getting this updated!

File: tools/xcodeproj_shims/install.sh
Comment (karim-alweheshy) [reply tools/xcodeproj_shims/install.sh:1]: why this change is necessary? here's the run without the change 

https://github.com/bazel-ios/rules_ios/actions/runs/7402381029/job/20140142409?pr=821

**Review by thiagohmcruz — COMMENTED** [review]
@karim-alweheshy thank you so much for tackling this and sorry about the huge delay here. I don't have all the context to review everything changing here but I'll try to do my part to unblock it. A few questions as a first pass.

File: tools/xcodeproj_shims/install.sh
Comment (karim-alweheshy) [reply tools/xcodeproj_shims/install.sh:1]: yes @thiagohmcruz 
the provided path is for a static lib while a dynamic library is what's expected

File: data_generators/xcspec_extractor.py
Comment (karim-alweheshy) [reply data_generators/xcspec_extractor.py:35]: yes exactly as you described it
the path is different in new Xcode version

File: rules/repositories.bzl
Comment (karim-alweheshy) [reply rules/repositories.bzl:344]: Because the old version was built for a different architecture which doesn't work anymore
The new artefact is build for the correct iOS architecture

File: tests/ios/unit-test/test-imports-app/BUILD.TensorFlowLiteC
Comment (karim-alweheshy) [reply tests/ios/unit-test/test-imports-app/BUILD.TensorFlowLiteC:9]: The new Xcode version can better detect and report misfitted architecture for the target builds
Part of the update is to point our third party dependencies to the right artefacts

File: tests/ios/xcconfig/BUILD.bazel
Comment (karim-alweheshy) [reply tests/ios/xcconfig/BUILD.bazel:90]: fixed in Xcode 15.2

File: tools/xcodeproj_shims/install.sh
Comment (thiagohmcruz) [reply tools/xcodeproj_shims/install.sh:1]: @karim-alweheshy do you know why this has become a problem now on Xcode 15? Or maybe this was only exposed now because you changed the third party archs? 

I just want to make sure we're not missing a potential edge case here that could break things after landing and understanding why exactly the change is necessary would help.

Comment (karim-alweheshy) [reply tools/xcodeproj_shims/install.sh:1]: I think we'll have to replace Tink with another framework built also for iOS simulator in mind
I'm trying to understand more why this has become a problem in Xcode 15.1 and not before

**Comment by luispadron** [comment]
@karim-alweheshy I'll rebase and merge this once #835 lands.

**Comment by luispadron** [comment]
Going to tackle this over in #837

**Pull request closed without merging**
